pkg/kubelet/kubelet.go

```diff
@@ -2112,6 +2132,25 @@ func (kl *Kubelet) syncLoopIteration(updates <-chan kubetypes.PodUpdate, handler
 			// TODO: Do we want to support this?
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
+	case e := <-plegCh:
```

Comment:
This sounds like an optimization. I'd prefer doing that later if you don't mind. Right now, it seems pretty fair to get only one event from the channel, when there are multiple channels in `select`. We can consider draining and batching some of them in the future.